packages/dd-trace/src/config.js

```diff
@@ -109,47 +120,14 @@ class Config {
     log.use(this.logger)
     log.toggle(this.debug, this.logLevel, this)
 
-    const DD_TRACING_ENABLED = coalesce(
-      process.env.DD_TRACING_ENABLED,
-      true
-    )
-    const DD_PROFILING_ENABLED = coalesce(
-      options.profiling, // TODO: remove when enabled by default
-      process.env.DD_EXPERIMENTAL_PROFILING_ENABLED,
-      process.env.DD_PROFILING_ENABLED,
-      false
-    )
-    const DD_PROFILING_EXPORTERS = coalesce(
-      process.env.DD_PROFILING_EXPORTERS,
-      'agent'
-    )
-    const DD_PROFILING_SOURCE_MAP = process.env.DD_PROFILING_SOURCE_MAP
-    const DD_RUNTIME_METRICS_ENABLED = coalesce(
-      options.runtimeMetrics, // TODO: remove when enabled by default
-      process.env.DD_RUNTIME_METRICS_ENABLED,
-      false
-    )
-    const DD_DBM_PROPAGATION_MODE = coalesce(
-      options.dbmPropagationMode,
-      process.env.DD_DBM_PROPAGATION_MODE,
-      'disabled'
-    )
-    const DD_DATA_STREAMS_ENABLED = coalesce(
-      options.dsmEnabled,
-      process.env.DD_DATA_STREAMS_ENABLED,
-      false
-    )
+    this.configWithOrigin = []
```

Comment:
This is already tracked elsewhere and shouldn't be tracked a second time.

Reply:
This is the object sent to app-started. Some [configuration values](https://github.com/DataDog/dd-trace-js/blob/f755fa1857e3e8fb66af5d0cb761f21a342fd59f/packages/dd-trace/src/config.js#L908-L934) are set using a mix of defaults, env vars and options, so we can't set them in the usual way. Since they are not changed through remote config or dynamic config as of now, I'm adding them at the end manually to `this.configWithOrigin` so they can be sent in app-started. We cannot manually add these values if we are only tracking config values using `changes`

Reply:
Not sure I understand. If the changes are already tracked when `merge` is called, and `merge` is called any time there is a change including when the app is started, then that data is already all available in telemetry. There is no need to track it a second time in the config object itself. That additional tracking can happen on the telemetry side from the existing changes.

Reply:
As in the [config values](https://github.com/DataDog/dd-trace-js/blob/f755fa1857e3e8fb66af5d0cb761f21a342fd59f/packages/dd-trace/src/config.js#L908-L934) whose origin is set to `calculated` cannot be set using `_applyDefaults`, `_applyEnvironment` and `_applyOptions`. So I'm keeping them the way they are and adding their origin in `_merge`. I can move [the addition of `calculated` origin](https://github.com/DataDog/dd-trace-js/blob/f755fa1857e3e8fb66af5d0cb761f21a342fd59f/packages/dd-trace/src/config.js#L925-L932) to the telemetry side, but it's using `_existsPropertyName` and `_getConfigValue` which are attached to the config object.

Reply:
This should all be done on the telemetry side. Why does telemetry need to access anything from the config object after the fact?

Reply:
Updated PR as we discussed. You can resolve this conversation if things seem good to you